src/cascadia/TerminalApp/ColorScheme.cpp

```diff
@@ -106,50 +100,56 @@ bool ColorScheme::ShouldBeLayered(const Json::Value& json) const
 // <none>
 void ColorScheme::LayerJson(const Json::Value& json)
 {
-    JsonUtils::GetValueForKey(json, NameKey, _schemeName);
-    JsonUtils::GetValueForKey(json, ForegroundKey, _defaultForeground);
-    JsonUtils::GetValueForKey(json, BackgroundKey, _defaultBackground);
-    JsonUtils::GetValueForKey(json, SelectionBackgroundKey, _selectionBackground);
-    JsonUtils::GetValueForKey(json, CursorColorKey, _cursorColor);
-
-    int i = 0;
-    for (const auto& current : TableColors)
+    JsonUtils::GetValueForKey(json, NameKey, _Name);
+    JsonUtils::GetValueForKey(json, ForegroundKey, _Foreground);
+    JsonUtils::GetValueForKey(json, BackgroundKey, _Background);
+    JsonUtils::GetValueForKey(json, SelectionBackgroundKey, _SelectionBackground);
+    JsonUtils::GetValueForKey(json, CursorColorKey, _CursorColor);
+
+    for (unsigned int i = 0; i < TableColors.size(); ++i)
     {
-        JsonUtils::GetValueForKey(json, current, _table.at(i));
-        i++;
+        JsonUtils::GetValueForKey(json, til::at(TableColors, i), _table.at(i));
     }
 }
 
-winrt::hstring ColorScheme::Name() const noexcept
+// Method Description:
+// - Update the given json object with values from this object.
+// Arguments:
+// - json: an object which will be a serialization of a ColorScheme object.
+// Return Value:
+// <none>
+void ColorScheme::UpdateJson(Json::Value& json)

```

Comment:
i 100% expected you to go a different direction with this and just .. make it a function called `ToJson` that returned the json object.

Reply:
Is there ever a reason for us to call UpdateJson _multiple times_ with different color schemes? (or profiles, etc.)?

Reply:
_Ideally_, I could name it `LayerJson` with an out param because we're really just layering our values onto the Json. But I guess that isn't something we're doing _yet_, so yeah, renaming to `ToJson` as you expected haha